Do something about the windows compilation warning: C++ designated initializers only available with '-std=c++20' or '-std=gnu++20' [-Wc++20-extensions]

On https://win-builder.r-project.org we get

```
* checking whether package 'RcppTskit' can be installed ... WARNING
Found the following significant warnings:
../inst/include/tskit/tskit/core.h:171:21: warning: C++ designated initializers only available with '-std=c++20' or '-std=gnu++20' [-Wc++20-extensions]
```

The code causing this is

```
/*
We define a specific NAN value for default mutation time which indicates
the time is unknown. We use a specific value so that if mutation time is set to
a NAN from a computation we can reject it. This specific value is a non-signalling
NAN with the last six fraction bytes set to the ascii of "tskit!"
*/
#define TSK_UNKNOWN_TIME_HEX 0x7FF874736B697421ULL
static inline double
__tsk_nan_f(void)
{
    const union {
        uint64_t i;
        double f;
    } nan_union = { .i = TSK_UNKNOWN_TIME_HEX };
    return nan_union.f;
}
```

The `.i =` is throwing the error, but only on CRAN Windows (not on GitHub windows or R Universe windows) and chat with tskit-devs suggests this is CRAN windows toolset issue https://github.com/tskit-dev/tskit/issues/3375. 

This is currently blocking CRAN submission :( #45

--- a/RcppTskit/src/Makevars.in
+++ b/RcppTskit/src/Makevars.in
@@ -59,6 +59,9 @@ RCPPTSKIT_LDFLAGS =
 PKG_CFLAGS = $(RCPPTSKIT_CFLAGS)
 PKG_CXXFLAGS = $(RCPPTSKIT_CXXFLAGS)
 
+# core.h:__tsk_nan_f requires C++20 on CRAN/Windows, but see #63
+CXX_STD = CXX20
+
 # Explicit compile rule for tskit C files
 tskit/%.o: tskit/%.c
 	$(CC) $(ALL_CPPFLAGS) $(PKG_CFLAGS) $(CFLAGS) $(CPICFLAGS) -c $< -o $@

